src/Service/Configurations/RuntimeConfigValidator.cs

```diff
@@ -239,25 +239,70 @@ public static void ValidateEntitiesDoNotGenerateDuplicateQueriesOrMutation(IDict
                 {
                     throw new DataApiBuilderException(
                         message: $"Entity {entityName} generates queries/mutation that already exist",
-                        statusCode: System.Net.HttpStatusCode.ServiceUnavailable,
+                        statusCode: HttpStatusCode.ServiceUnavailable,
                         subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError);
                 }
             }
         }
 
         /// <summary>
-        /// Check whether the entity name defined in runtime config only contains
-        /// characters allowed for GraphQL names.
-        /// Does not perform validation for entities which do not
+        /// Check whether the entity configuration defined in runtime config only contains characters allowed for GraphQL names
+        /// and other validations related to rest path and methods configured for the entity.
+        /// The GraphQL validation is not performed for entities which do not
         /// have GraphQL configuration: when entity.GraphQL == false or null.
         /// </summary>
         /// <seealso cref="https://spec.graphql.org/October2021/#Name"/>
-        /// <param name="runtimeConfig"></param>
-        public static void ValidateEntityNamesInConfig(Dictionary<string, Entity> entityCollection)
+        /// <param name="runtimeConfig">The runtime configuration.</param>
+        public static void ValidateEntityConfiguration(RuntimeConfig runtimeConfig)
```

Comment:
Do you need the complete runtime config as input to this function? Seems like only the entity name value pairs is what you need, so the previous input Dictionary seems to not required to change.

Reply:
Properties like runtimeConfig.RestGlobalSettings.Enabled and runtimeConfig.GraphQLGlobalSettings.Enabled are needed. I can instead use the existing dictionary parameter but then I would need two different methods(and loops) for GQL and REST validations. So, I thought I'll change to use runtimeConfig which provides flexibility for any potential future dependencies on values in the config outside the entity's configuration.